### PR TITLE
Fix tck framework

### PR DESCRIPTION
**Pull request by DvirDukhan**
enabled TCK testing framework to the new result set structure

### CODE REVIEW

File: src/resultset/formatters/resultset_replycompact.c [hunk 1]
```diff
@@ -163,7 +163,7 @@ void ResultSet_ReplyWithCompactHeader(RedisModuleCtx *ctx, const ResultSetHeader
         RedisModule_ReplyWithArray(ctx, 2);
         Column *c = header->columns[i];
         ColumnTypeUser t;
-        char *identifier = c->alias? c->alias: c->name;
+        char *identifier = c->name;
```
Comment (swilly22) [inline src/resultset/formatters/resultset_replycompact.c]: Use alias when provided as identifier for line 183, lookup entity using c->name.

File: tests/tck/utils/assertions.py [hunk 2]
```diff
@@ -1,10 +1,76 @@
...
+    """ Returns True if string is a number. """
+    if not isinstance(s, (Number, basestring)):
+        return False
     try:
         float(s)
         return True
     except ValueError:
         return False
+    
+def prepareActualValue(actualValue):
```
Comment (swilly22) [inline tests/tck/utils/assertions.py]: Please document each function.

File: tests/tck/utils/assertions.py [hunk 3]
```diff
@@ -1,10 +1,76 @@
...
+        actualValue = prepareString(actualValue)
```
Comment (swilly22) [inline tests/tck/utils/assertions.py]: This seems like each function will overwrite the previous one,
my guess without reading the implementation of prepareString, stringToNumeric, nodeToString and edgeToString
is that they do nothing if actualValue is set.

Please avoid the unnecessary calls by returning once actualValue been set.

File: tests/tck/utils/assertions.py [hunk 4]
```diff
@@ -1,10 +1,76 @@
...
+def prepareExpectedValue(expectedValue):
+    # string preparation
+    expectedValue = prepareString(expectedValue)
+    if expectedValue == "true":
+        expectedValue = True
+    if expectedValue == "false":
+        expectedValue = False
+    if expectedValue == "null":
+        expectedValue = None
+    expectedValue = stringToNumeric(expectedValue)
```
Comment (swilly22) [inline tests/tck/utils/assertions.py]: Do we really need to call stringToNumeric if we entered any of the conditions above?

File: tests/tck/utils/assertions.py [hunk 5]
```diff
@@ -1,10 +1,76 @@
...
+    return expectedValue
+
+def prepareString(value):
```
Comment (swilly22) [inline tests/tck/utils/assertions.py]: I would have rename it to something like: "remote quotes"

File: tests/tck/utils/assertions.py [hunk 6]
```diff
@@ -1,10 +1,76 @@
...
+    if isinstance(value, basestring):
+        value = value.replace("'", "")
+        value = value.replace('"', "")
+    return value
+
+def stringToNumeric(value):
+    if is_number_tryexcept(value):
```
Comment (swilly22) [inline tests/tck/utils/assertions.py]: Is it OK for this function to return `false`?
can't is_number_tryexcept return the number represented by value?

File: tests/tck/utils/assertions.py [hunk 7]
```diff
@@ -1,10 +1,76 @@
...
+        value = float(value)
+        if value.is_integer():
+            value = int(value)
+    return value
+
+def nodeToString(value):
+    if isinstance(value, Node):
```
Comment (swilly22) [inline tests/tck/utils/assertions.py]: what if value is not a Node object the function will return the original value.

File: tests/tck/utils/assertions.py [hunk 8]
```diff
@@ -1,10 +1,76 @@
...
+        if value.properties:
+            props = ','.join(key+': '+str(val) for key, val in value.properties.items())
+            if value.label:
+                res+=" "
+            res += '{' + props + '}'
+        res += ')'
+        value = res
+    return value
+
+def edgeToString(value):
```
Comment (swilly22) [inline tests/tck/utils/assertions.py]: what if value is not an Edge object the function will return the original value.

File: tests/tck/utils/assertions.py [hunk 9]
```diff
@@ -31,6 +98,7 @@ def assert_resultset_length(resultset, length):
     assert(len(resultset.result_set) == length)
 
 def assert_resultset_content(resultset, expected):
+    resultset.pretty_print()
```
Comment (swilly22) [inline tests/tck/utils/assertions.py]: Why are we printing here?

File: tests/tck/utils/assertions.py [hunk 10]
```diff
@@ -40,28 +108,6 @@ def assert_resultset_content(resultset, expected):
...
-                if expectedCell.is_integer():
-                    expectedCell = int(expectedCell)
-
-            # if actualCell is not expectedCell:
-            #     print "actualCell: %s differ from expectedCell: %s\n" % (actualCell, expectedCell)
-            # else:
-            #     print "PERFECTO!\n"
-            assert actualCell == expectedCell, "actualCell: %s differ from expectedCell: %s\n" % (actualCell, expectedCell)
+            actualValue = prepareActualValue(actualRow[cellIdx])
+            expectedValue= prepareExpectedValue(expectedRow[cellIdx])
```
Comment (swilly22) [inline tests/tck/utils/assertions.py]: missing space before `=`

File: src/resultset/formatters/resultset_replycompact.c (hunk 1 above)
Comment (DvirDukhan) [reply src/resultset/formatters/resultset_replycompact.c]: done

File: tests/tck/utils/assertions.py (hunk 10 above)
Comment (DvirDukhan) [reply tests/tck/utils/assertions.py]: fixed

File: tests/tck/utils/assertions.py (hunk 9 above)
Comment (DvirDukhan) [reply tests/tck/utils/assertions.py]: removed

File: tests/tck/utils/assertions.py (hunk 5 above)
Comment (DvirDukhan) [reply tests/tck/utils/assertions.py]: done

File: tests/tck/utils/assertions.py (hunk 4 above)
Comment (DvirDukhan) [reply tests/tck/utils/assertions.py]: done

File: tests/tck/utils/assertions.py (hunk 7 above)
Comment (DvirDukhan) [reply tests/tck/utils/assertions.py]: fixed

File: tests/tck/utils/assertions.py (hunk 8 above)
Comment (DvirDukhan) [reply tests/tck/utils/assertions.py]: fixed

File: tests/tck/utils/assertions.py (hunk 6 above)
Comment (DvirDukhan) [reply tests/tck/utils/assertions.py]: modifed

File: tests/tck/utils/assertions.py (hunk 3 above)
Comment (DvirDukhan) [reply tests/tck/utils/assertions.py]: fixed

File: tests/tck/utils/assertions.py (hunk 2 above)
Comment (DvirDukhan) [reply tests/tck/utils/assertions.py]: done

**Pull request merged**